src/ipc_listener.cpp

```diff
@@ -208,9 +260,13 @@ int zmq::ipc_listener_t::set_address (const char *addr_)
 
 int zmq::ipc_listener_t::close ()
 {
-    zmq_assert (s != retired_fd);
-    int rc = ::close (s);
-    errno_assert (rc == 0);
+    int rc;
+
+    if ( s != -1 ) {
+      zmq_assert (s != retired_fd);
```

Comment:
On Mon, Mar 14, 2016 at 11:10 AM, Luca Boccassi
<notifications@github.com> wrote:
> If you need more time, I can merge as-is for now, and then you can send another PR to fix it when it's ready. Or maybe it's easier for you to work on the same branch?

I can wait until I get things into a shape that meets with your
approval.  No point in checking in "sorta" code.

> I personally don't have any objections to exceptions, but it's true that they are not used anywhere else. @hintjens any objections/opinions on exceptions?

Well, if they aren't commonly used, it probably isn't that helpful to
divert from the common style already in use.  I'll rework it without
exceptions.

Pete